src/handlers/int-test-support/helpers/testSetup.ts

```diff
@@ -6,51 +6,69 @@ import {
   listS3Objects,
 } from "./s3Helper";
 
-
 const prefix = resourcePrefix();
 const storageBucket = `${prefix}-storage`;
+const rawInvoiceBucket = `${prefix}-raw-invoice`;
+const rawInvoiceTextractBucket = `${prefix}-raw-invoice-textract-data`;
 
 export default async function globalSetup(): Promise<void> {
-  // Delete objects with prefix "btm_event_data/2005"
-  await deleteS3ObjectsByPrefix({
-    bucket: storageBucket,
-    prefixes: ["btm_event_data/2005"],
-  });
-  // poll to ensure that the objects with prefix "btm_event_data/2005" have been deleted
+  console.log("Cleaning up the environment before test execution starts");
+  await deleteS3ObjectsAndPoll(storageBucket, "btm_event_data/2005");
+  await deleteS3ObjectsAndPoll(storageBucket, "btm_invoice_data");
+  await deleteS3ObjectsAndPoll(storageBucket, "btm_invoice_data_archived");
+  await deleteAllObjects(rawInvoiceBucket);
+  await deleteAllObjects(rawInvoiceTextractBucket);
+  console.log("All previous test data has been cleaned.");
+}
+
+const deleteS3ObjectsAndPoll = async (
+  bucketName: string,
+  prefix: string
+): Promise<void> => {
+  await deleteS3ObjectsByPrefix({ bucket: bucketName, prefixes: [prefix] });
+
+  // poll to ensure that the objects with prefix have been deleted
   await poll(
     async () =>
       await listS3Objects({
-        bucketName: storageBucket,
-        prefix: "btm_event_data/2005",
+        bucketName,
+        prefix,
       }),
     (s3Objects) => s3Objects.length === 0,
     {
       timeout: 60000,
       interval: 10000,
-      notCompleteErrorMessage:
-        "btm_event_data/2005 folder could not be deleted because it still contains objects",
+      notCompleteErrorMessage: `${prefix} folder could not be deleted because it still contains objects`,
```

Comment:
If the query doesn't complete, is it guaranteed to be because it still contains objects?

Reply:
the poll check is checking the length of the s3 folder and if the length is not zero, it implies that poll failed because there are still objects in the folder